### PR TITLE
Changes javascript code to use react 

### PR DESCRIPTION
**Pull request by mercebc**

### CODE REVIEW

File: assets/js/locations.js [hunk 1]
```diff
@@ -1,41 +1,16 @@
...
-
-const checkResultContent = (result) => {
-  result.count > 0 ? renderResult(result) : renderNotFound();
+import * as React from 'react'
+import * as ReactDOM from 'react-dom'
+
+import SearchForm from './search-form'
+import request from './request'
+
+export default function Locations(props){
```
Comment (ssunday) [inline assets/js/locations.js]: I personally prefer no default exports if possible

File: assets/js/results-list.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import * as React from 'react'
+import * as ReactDOM from 'react-dom'
+
+export function ResultsTitle(props){
+    return(
+      <p className="results-title">Results for {props.city}</p>
+    );
+}
+
+export function SingleResult(props){
```
Comment (ssunday) [inline assets/js/results-list.js]: usually I see this as export const SingleResult = (props) => {}

File: assets/js/search-form.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+      .catch(error => this.setState({resultsError: true}))
+  }
+
+  resultsList(){
+    if(this.state.resultsError){
+        return <ServerError/>
+    }else if (this.state.resultsList !== null){
+      if(this.state.resultsList.count > 0) {
+        return <ResultsList locations={this.state.resultsList}></ResultsList>
+      }else{
```
Comment (ssunday) [inline assets/js/search-form.js]: spacing off

File: assets/js/results-list.js (hunk 2 above)
Comment (mercebc) [reply assets/js/results-list.js]: would that mean that I have to change how the render() function is written as well? To be consistent with the version of javascript that I am using?

File: assets/js/locations.js (hunk 1 above)
Comment (mercebc) [reply assets/js/locations.js]: Thanks! Applied!!!

File: assets/js/results-list.js (hunk 2 above)
Comment (ssunday) [reply assets/js/results-list.js]: no, you'd just change the definition of the component

**Pull request merged**